### ISSUE TITLE
Wrap code as package

### ISSUE THREAD

**Issue by wishcow79** [wishcow79/chabauty#2]
Wrap the chabauty code as a Magma package
http://magma.maths.usyd.edu.au/magma/handbook/text/24

**Comment by koffie** [comment]
Yes, this is a good idea, maybe this issue can be subdivided in several related ones.

1. Put the functions that a user should not care about in a different file
2. Use the {comment text} option of intrinsics to write documentation for functions that might be usefull to other.
3. ?
4. Profit!!!
